californium-core/src/main/java/org/eclipse/californium/core/network/Exchange.java

```diff
@@ -466,8 +489,9 @@ public boolean isComplete() {
 	public void setComplete() {
 		this.complete = true;
 		ExchangeObserver obs = this.observer;
-		if (obs != null)
+		if (obs != null) {
```

Comment:
is there a particular reason why we do not simply invoke `completeCurrentRequest` at this point?